### PR TITLE
New dependencies layout

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: cmake/ProjectJsonCpp.cmake [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    URL https://github.com/open-source-parsers/jsoncpp/archive/1.7.7.tar.gz
+    URL_HASH SHA256=087640ebcf7fbcfe8e2717a0b9528fff89c52fcf69fa2a18cc2b538008098f97
+    CMAKE_COMMAND ${JSONCPP_CMAKE_COMMAND}
+    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+               # Build static lib but suitable to be included in a shared lib.
+               -DCMAKE_POSITION_INDEPENDENT_CODE=${BUILD_SHARED_LIBS}
+               -DJSONCPP_WITH_TESTS=Off
+               -DJSONCPP_WITH_PKGCONFIG_SUPPORT=Off
+    LOG_CONFIGURE 1
+    # Overwtire build and install commands to force Release build on MSVC.
```
Comment (chriseth) [inline cmake/ProjectJsonCpp.cmake]: Typo :-)

File: cmake/scripts/helpers.cmake [hunk 2]
```diff
@@ -0,0 +1,90 @@
+get_filename_component(ROOT_DIR "${CMAKE_CURRENT_LIST_DIR}/../.." ABSOLUTE)
+
+set(CACHE_DIR "${ROOT_DIR}/deps/src")
+set(PACKAGES_DIR "${ROOT_DIR}/packages")
+
+function(download URL DST_FILE STATUS)
```
Comment (chriseth) [inline cmake/scripts/helpers.cmake]: Why does the functionality offered by cmake not suffice here?

Comment (chfast) [inline cmake/scripts/helpers.cmake]: To be honest, I don't remember. It was my first contact with CMake scripting (this file is copied from cpp-dependencies, only needed by cpp-ethereum on Windows). It looks I added some kind of checking if the file is already there. ExternalProject does it nicely, but pure `file(DOWNLOAD ...)` maybe not.

**Comment by chriseth** [comment]
Ok, good to go from my side.

**Comment by chfast** [comment]
Ok. Let's spin it up.

**Pull request merged**